### PR TITLE
Normalize exception domain names for Content Blocking

### PR DESCRIPTION
**Pull request by bwaresiak**
Task/Issue URL: https://app.asana.com/0/856498667320406/1202191081651325

**Description**:
Normalize exception domain names for Content Blocking when adding these from Unprotected Sites screen.

**Steps to test this PR**:

Test disabling protections on websites with non-ascii characters in address (punycoded).

Test unprotected sites on Privacy Test pages.

Unprotected sites list:
 - Tweak `AppPrivacyConfiguration:userDisabledProtection(_:)` to disable normalization when adding new domains.
 - Go to unprotected sites list in Settings.
 - Add 'Example.com'
 - Navigate to example.com - check protection status.
 - Re-enable protection from Privacy Dashboard.

**Device Testing**:

* [ ] iPhone 
* [ ] iPad

**OS Testing**:

* [ ] iOS 13
* [ ] iOS 14
* [ ] iOS 15

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**

### CODE REVIEW

File: Core/StringExtension.swift [hunk 1]
```diff
@@ -19,6 +19,7 @@
 
 import Foundation
 import Punycode
```
Comment (Bunn) [inline Core/StringExtension.swift]: I believe we can remove the Punnycode import.

**Review by Bunn — APPROVED** [review]
✅ Migrating from dev branch with unprotected Example.com, it was correctly migrated to example.com
✅ Changing the code to allow unsanitized URLs and then testing unprotected sites worked correctly
✅ Non-ascii characters are correctly being sanitized when saving
✅ Enabling/Disabling protection works correctly on the privacy test page.

Besides a small import that looks unnecessary everything LGTM!

**Review by Bunn — APPROVED** [review]
✅ Punnycode import changed to BSK
✅ BSK bumped to 12.2.8 from 12.2.7
✅ Tested adding sites like Example.com to unprotected list then opening them
✅ Tested opening sites with emoji on it then disabling/enabling protection

LGTM

**Comment by bwaresiak** [comment]
Thank you @Bunn

**Pull request merged**